src/main/java/frc/robot/subsystems/climber/ClimberIOTalonFX.java

```diff
@@ -0,0 +1,268 @@
+package frc.robot.subsystems.climber;
+
+import static frc.robot.subsystems.climber.ClimberConstants.*;
+
+import com.ctre.phoenix6.BaseStatusSignal;
+import com.ctre.phoenix6.StatusCode;
+import com.ctre.phoenix6.StatusSignal;
+import com.ctre.phoenix6.configs.CurrentLimitsConfigs;
+import com.ctre.phoenix6.configs.TalonFXConfiguration;
+import com.ctre.phoenix6.controls.PositionTorqueCurrentFOC;
+import com.ctre.phoenix6.controls.PositionVoltage;
+import com.ctre.phoenix6.controls.TorqueCurrentFOC;
+import com.ctre.phoenix6.controls.VelocityTorqueCurrentFOC;
+import com.ctre.phoenix6.controls.VoltageOut;
+import com.ctre.phoenix6.hardware.TalonFX;
+import com.ctre.phoenix6.signals.InvertedValue;
+import com.ctre.phoenix6.signals.NeutralModeValue;
+import frc.lib.team3015.subsystem.FaultReporter;
+import frc.lib.team3061.RobotConfig;
+import frc.lib.team3061.drivetrain.swerve.Conversions;
+import frc.lib.team6328.util.Alert;
+import frc.lib.team6328.util.Alert.AlertType;
+import frc.lib.team6328.util.TunableNumber;
+
+/** TalonFX implementation of the generic SubsystemIO */
+public class ClimberIOTalonFX implements ClimberIO {
+  private TalonFX leftMotor;
+  private TalonFX rightMotor;
+
+  private PositionTorqueCurrentFOC leftPositionCurrentRequest;
```

Comment:
We should try to use Motion Magic Expo Torque Current FOC control. That's a mouthful!

Reply:
https://pro.docs.ctr-electronics.com/en/latest/docs/api-reference/device-specific/talonfx/motion-magic.html#using-motion-magic-expo-in-api